### PR TITLE
adds obscape reader

### PR DESCRIPTION
**Pull request by RubendeBruin**
Added reader for obscape buoy CSV files

Test-files still to be added (for OBScape to supply)

link: https://obscape.com/site/

### CODE REVIEW

**Comment by rafa-guedes** [comment]
Thanks for the new reader  @RubendeBruin , I'll wait for the sample test file to be able to check this before merging ok?

**Comment by RubendeBruin** [comment]
Hi @rafa-guedes , I've added two test files, a test function and included the reader in the docs.
I've kept the testing quite basic, we have heavily used the function over the past two months without any issues.

Ready to be merged.

**Comment by rafa-guedes** [comment]
Sounds good @RubendeBruin thanks for that. I'll try and merge this in the next few days alongside https://github.com/wavespectra/wavespectra/pull/106 which is currently under review to be released as v4.0.0 soon.

**Comment by rafa-guedes** [comment]
@RubendeBruin there are several broken tests in this pull request due to replacing the `.dir` attribute by `.direction`. Whilst I agree direction is more declarative this is a longstanding convention in the library which will cause problems. Do you mind undoing these changes please?

File: wavespectra/input/triaxys.py
Comment (RubendeBruin) [inline wavespectra/input/triaxys.py:1]: .show()  is what we want, ok

File: wavespectra/output/octopus.py
Comment (RubendeBruin) [inline wavespectra/output/octopus.py:1]: Think it is good to explicitly close the file

File: tests/io/test_ndbc.py
Comment (RubendeBruin) [inline tests/io/test_ndbc.py:1]: I think the assert needs to be here

**Review by RubendeBruin — COMMENTED** [review]
looks ok now

**Comment by RubendeBruin** [comment]
@hi @rafa-guedes , I do not remember intentionally changing .dir to .direction but indeed it did happen. Did a replace all and from the diff all seems ok now. Somewhere in the last few years the branches must have diverged.

I did notice a few other minor changes. I commented on those in the review above.

**Comment by rafa-guedes** [comment]
@RubendeBruin I have forked your obscape branch to make a few changes and and then open a pull request to merge onto yours - I didn't realise though my commits went straight to your branch instead of my fork. So my changes are already here :|

Anyways, the main thing is a change to the arguments in the `read_obscape` function. It is a bit opinionated / risky to have those functions rely in the file name pattern, and this goes against the behaviour of all the other readers in wavespectra. So I changed `read_obscape` to accept `filename_or_fileglob` instead, similar to other readers such as `read_triaxys`. `filename_or_fileglob` can be the path name for a file, a list of path names, or a glob string which is expanded. I kept your old behaviour to read from a directory based on start/end dates in a new function, `read_obscape_dir` (which will need to be imported using the full module path).

Let me know if you are happy with these changes and I'll merge.

**Comment by RubendeBruin** [comment]
Hi @rafa-guedes , this change works fine for me. Good to keep things consistent.
Happy with the changes so feel free to merge.

sidenote:
One thing that I did note is that your updated version stores the time-vector as `pandas._libs.tslibs.timestamps.Timestamp` objects whereas the version that I had stored it as `datetime64[ns]`. But that change seems to originate from outside the reader function. Should have nothing to do with the replacement of `datetime.utcfromtimestamp(int(timestamp))` to `datetime.fromtimestamp(int(timestamp), tz=timezone.utc)` as both return a datetime object.

**Comment by rafa-guedes** [comment]
Not sure about that one, perhaps it could be due to changes in pandas / xarray versions? I replaced utcfromtimestamp because it was deprecated, I got a deprecation message in my environment but as you said, the change should have returned the same object.

**Comment by RubendeBruin** [comment]
did some tests and the difference does come from replacing utcfromtimestamp with fromtimestamp. Apparently xarray converts datetime without tz info to datetime64 and datetime with tz to pandas timestamps.

```
utcfromtimestamp --> 
   datetime.datetime(2024, 4, 3, 9, 30) -> 
       <class 'numpy.datetime64'>

fromtimestamp --> 
    datetime.datetime(2024, 4, 3, 9, 30, tzinfo=datetime.timezone.utc) -> 
         <class 'pandas._libs.tslibs.timestamps.Timestamp'>
```

I like the pandas timestamps better so for me this is an improvement.

**Comment by rafa-guedes** [comment]
Ok well spotted. I think we should actually remove the timezone info, because this sometimes causes some issues when using the data together with other data that are not timezone aware..

**Comment by RubendeBruin** [comment]
ok, fixed that.
also added a test to make sure that the time has the correct value and type

**Pull request merged**